### DISCUSSION TITLE
FormData body together with  'Content-Type': 'multipart/form-data'

### DISCUSSION THREAD

**Discussion by capaj** [node-fetch/node-fetch#1703] [Ideas]
When you call fetch like this:

```ts
import fetch from 'node-fetch'
import { FormData } from 'formdata-node'
import { Readable } from 'stream'

const formData = new FormData()
const stream = Readable.from(buffer)
formData.append('file', new BlobFromStream(stream, data.length), fileName)

fetch(URL, {
    headers: {
         'Content-Type': 'multipart/form-data'
    },
    method: 'POST',
    body: formData
  })
```

it does not log a warning, it does not throw any error, yet it does not send ANY body. The request is sent with empty body.
Why?

Also I ran into a [claim](https://stackoverflow.com/questions/46640024/how-do-i-post-form-data-with-fetch-api#comment93550661_46640744) that when you do this in a browser, that the browser just ignores the header and sends the body as expected.
If that is the case, maybe node-fetch should override the header in this case too to align with how browsers deal with this weird situation?

If maintainers agree, I would love to open an issue for this and maybe even a PR.

**Comment by jimmywarting** [comment]
you don't need to import `formdata-node` node-fetch comes with formdata builtin cuz it's part of the specification to be able to parse bodies using `await body.formData()` so you can just import it using `import fetch, {FormData} from 'node-fetch'`

secondly, you should not be creating a stream from a buffer and then convert it to a blob, just create a blob directly with the buffer insteand of using BlobFromStream

and for the third: you should not specify the content-type yourself when posting a formdata as that is being done for you automatically. it is neccessary to omit this cuz we also need to add the boundary key/value meta in the content-type header
setting content-type header yourself should be avoided. 

so all in all:

```js
import fetch, { FormData, File } from 'node-fetch'

const formData = new FormData()
const file = new File([buffer], filename)
formData.append('file', file)

const response = await fetch(URL, {
  method: 'POST',
  body: formData
})
const result = await response.arrayBuffer()
```

**Reply by capaj** [reply]
Thanks. I know I was not using the API correctly, but thanks for showing me the correct way.
The code I previously had is something which can be found in many stack overflow posts.

I was asking if it would be ok to align node-fetch with how the browsers handle things. So the question remains unanswered.

**Reply by jimmywarting** [reply]
what exactly would you like to change to align it more with web?
